images/airflow/2.9.2/python/mwaa/celery/task_monitor.py

```diff
@@ -70,6 +72,14 @@
 # consumption, we reset the worker to non-idle state and backoff from checking for
 # idleness for a minute.
 IDLENESS_RESET_BACKOFF_PERIOD = timedelta(minutes=1)
+# In case of issues, signals can arrive late and out of order. So, we scan all unprocessed signals from the last one hour.
+# Only a handful of signals are expected for each worker, so this repeated processing should be very light.
+SIGNAL_SEARCH_TIME_RANGE = timedelta(hours=1)
+# If a worker activation signal has not been received in the first 10 minute. Then, we give up on waiting anymore for the signal.
+ACTIVATION_WAIT_TIME_LIMIT = timedelta(minutes=10)
```

Comment:
Should these all be defaults that can be overridden with Environment Variables?

Reply:
We can do that. However, there are a lot of these variables and I am not sure of the value of adding the additional logic. If the variables are introduced, in order to change the behavior, the value of the variable will have to be changed which would still require an environment update.